src/AspNetCore/Src/proxymodule.cxx

```diff
@@ -79,32 +78,87 @@ CProxyModule::OnExecuteRequestHandler(
     IHttpEventProvider *
 )
 {
-    m_pHandler = new FORWARDING_HANDLER(pHttpContext);
-    if (m_pHandler == NULL)
+    HRESULT hr;
+    APPLICATION_MANAGER* pApplicationManager;
+    APPLICATION* pApplication;
+    ASPNETCORE_CONFIG* config;
+    ASPNETCORE_APPLICATION* pAspNetCoreApplication;
+    ASPNETCORE_CONFIG::GetConfig(pHttpContext, &config);
+
+    // TODO store whether we are inproc or outofproc so we don't need to check the config everytime?
+    if (config->QueryHostingModel()->IsEmpty() || config->QueryHostingModel()->Equals(L"outofprocess", TRUE))// case insensitive
     {
-        pHttpContext->GetResponse()->SetStatus(500, "Internal Server Error", 0, E_OUTOFMEMORY);
-        return RQ_NOTIFICATION_FINISH_REQUEST;
+        m_pHandler = new FORWARDING_HANDLER(pHttpContext);
+        if (m_pHandler == NULL)
+        {
+            hr = E_OUTOFMEMORY;
+            goto Failed;
+        }
+
+        return m_pHandler->OnExecuteRequestHandler();
     }
-
-    return m_pHandler->OnExecuteRequestHandler();
+    else if (config->QueryHostingModel()->Equals(L"inprocess", true))
+    {
+        pApplicationManager = APPLICATION_MANAGER::GetInstance();
+        if (pApplicationManager == NULL)
+        {
+            hr = E_OUTOFMEMORY;
+            goto Failed;
+        }
+
+        hr = pApplicationManager->GetApplication(pHttpContext,
+            &pApplication);
+        if (FAILED(hr))
+        {
+            goto Failed;
+        }
+
+
```

Comment:
Add a static counter such that we don't try to create the application more than ~5 times.